lib/knigge/options.ex

```diff
@@ -128,8 +128,8 @@ defmodule Knigge.Options do
       opts
       |> map_deprecated()
       |> validate!()
-      |> with_defaults()
       |> transform(with_env: env)
+      |> with_defaults()
 
```

Comment:
What's the train of thought here of swapping these two? This would actually break the `delegate_at_runtime?` default, since the "environment" spec gets resolved to a boolean in `transform/2`.